backend/src/apiserver/server/visualization_server.go

```diff
@@ -115,13 +112,15 @@ func (s *VisualizationServer) getVisualizationServiceURL(request *go_client.Crea
 	return s.serviceURL
 }
 
-func isVisualizationServiceAlive(serviceURL string) bool {
+func isVisualizationServiceAlive(serviceURL string) error {
 	resp, err := http.Get(serviceURL)
+
 	if err != nil {
-		glog.Error("Unable to verify visualization service is alive!", err)
-		return false
+		return util.Wrap(err, "Internal Error: the visualization server host not exist!")
```

Comment:
could you both logging and returning the errors. Sometimes, when we debug, we also look up the log for details. So we probably should have the error reflected in both the returns and in the logs

Reply:
Other than that /lgtm

Reply:
+1
and did you forget to update the error message?

Reply:
Good points! Will fix in new revision of PR